src/serverlist.cpp

```diff
@@ -415,20 +425,19 @@ void CServerListManager::CentralServerQueryServerList ( const CHostAddress& Inet
             {
                 // check if the address of the client which is requesting the
                 // list is the same address as one server in the list -> in this
-                // case he has to connect to the local host address
+                // case he has to connect to the local host address and port
+                // to allow for NAT.
                 if ( vecServerInfo[iIdx].HostAddr.InetAddr == InetAddr.InetAddr )
                 {
-                    vecServerInfo[iIdx].HostAddr.InetAddr =
-                        QHostAddress ( QHostAddress::LocalHost );
```

Comment:
Ok, so no "LocalHost" required anymore since if we are running on the same PC, the local IP address should also work, right?

Reply:
It should, yes.  Even I can't think of when a server would not route its own IP to itself.